### PR TITLE
Support Pydantic V2

### PR DESCRIPTION
**Pull request by ahans**
This updates the code to be compatible with Pydantic V2. We also introduce a smoke test for importing `kiss_icp`. It's still just a smoke test, but does a bit more than only calling `kiss_icp_pipeline --version`. The latter would also pass with Pydantic V2 being installed, while the import would fail.

### CODE REVIEW

**Comment by ahans** [comment]
This isn't quite ready yet.

File: python/kiss_icp/config/parser.py
Comment (ahans) [inline python/kiss_icp/config/parser.py:44]: Accessing an attribute before the parent's constructor was called is not possible with Pydantic V2. But even with the previous version, the pattern here is not considered very "pydantic" (see, e.g., [here](https://stackoverflow.com/a/65873630/250826)). Having the model class as plain as possible and filling it from the outside seems to be cleaner.

**Comment by nachovizzo** [comment]
@ahans I'm more than happy to see you back in the PR section. Thanks a lot!!! I'm checking this now

File: .github/workflows/python.yml
Comment (nachovizzo) [inline .github/workflows/python.yml:28]: I will see this on the CI and pretend we have unit tests 😜 Just won't open the tab

File: python/kiss_icp/config/parser.py
Comment (nachovizzo) [reply python/kiss_icp/config/parser.py:44]: Thanks for clarifying this!

**Review by nachovizzo — APPROVED** [review]
Great! Happy to upgrade to the new pydantic version! Thanks for leading the effort

Waiting on @benemer for approval + merge

**Review by benemer — APPROVED** [review]
All good from my side, thanks for the support :)

**Comment by ahans** [comment]
@nachovizzo thanks for catching my cleanup failures!

**Pull request merged**
